examples/http.rs

```diff
@@ -0,0 +1,116 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use hyper::header::CONTENT_TYPE;
+use serde::{Deserialize, Serialize};
+
+use esp_idf_svc::hal::prelude::Peripherals;
+use esp_idf_svc::log::EspLogger;
+use esp_idf_svc::wifi::{BlockingWifi, EspWifi};
+use esp_idf_svc::{eventloop::EspSystemEventLoop, nvs::EspDefaultNvsPartition};
+
+use lynx_embedded::{reqwesp, wifi as espWifi};
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug)]
```

Comment:
What does this do?

Reply:
`#[allow(dead_code)]` is a typing hint that stops the compiler from spitting out a warning if some of the fields of this struct are never used. It's useful if you want to implement fields that you intend to use later.

The `#[derive(Deserialize, Debug)]` macro attempts to automatically implement (`impl`) the `Deserialize` and `Debug` traits.
[`Debug`](https://doc.rust-lang.org/rust-by-example/hello/print/print_debug.html) a built-in trait that is required for a struct to be printable.
`Deserialize` is a trait from the `serde` crate. A struct must implement this trait in order for json to be converted into this struct. Similarly, the `Serialize` trait is required to convert a struct into json.